manifests/producer/service.pp

```diff
@@ -25,10 +26,15 @@
     fail('[Producer] You need to specify a value for topic')
   }
 
-  file { '/etc/init.d/kafka-producer':
-    ensure  => present,
-    mode    => '0755',
-    content => template('kafka/producer.init.erb'),
+  if $::service_provider == 'systemd' {
+    fail('Console Producer is not supported on systemd, because the stdin of the process cannot be redirected')
```

Comment:
I'm not exactly sure about the root of the issue. I think that should be possible. Can we discuss that in IRC?